test/libsolidity/ASTJSON/assembly/switch.json

```diff
@@ -157,7 +157,23 @@
                     }
                   ]
                 },
-                "externalReferences": [],
+                "externalReferences":
```

Comment:
Not sure why this test changed, but having more resolved external references is a good thing, right?

Reply:
hmm. .this is the source for this AST

```
contract C {
    function f() pure public {
        assembly {
            let f := 0
            switch calldatasize()
            case 0 { f := 1 }
            default { f := 2 }
        }
    }
}
```
maybe it's related to the fact that the function and the local variable both are named `f`. Investigating.

Reply:
okay, so the id 5 that is mentioned here is def. the `f()` function..

Reply:
Shouldn't this be caught by shadowing rules?

Reply:
When you do a normal compilation it is caught, yes. But not for the ASTJSON test..

Reply:
Alright, the previous code that used the AsmAnalyzer checked if the variable was declared in scope before invoking the ext. resolver which this visitor didn't do.

I fixed it by adding my own simple scope tracking.